apps/oxlint/src/output_formatter/stylish.rs

```diff
@@ -0,0 +1,146 @@
+use std::io::Write;
+
+use oxc_diagnostics::{
+    reporter::{DiagnosticReporter, Info},
+    Error, Severity,
+};
+use rustc_hash::FxHashMap;
+
+use crate::output_formatter::InternalFormatter;
+
+#[derive(Debug, Default)]
+pub struct StylishOutputFormatter;
+
+impl InternalFormatter for StylishOutputFormatter {
+    fn all_rules(&mut self, writer: &mut dyn Write) {
+        writeln!(writer, "flag --rules with flag --format=stylish is not allowed").unwrap();
+    }
+
+    fn get_diagnostic_reporter(&self) -> Box<dyn DiagnosticReporter> {
+        Box::new(StylishReporter::default())
+    }
+}
+
+#[derive(Default)]
+struct StylishReporter {
+    diagnostics: Vec<Error>,
```

Comment:
can this be `FxHashMap<String, Vec<&Error>>` and the grouping will be done inside `render_error` :)?